test/reducers/users.spec.js

```diff
@@ -0,0 +1,79 @@
+import deepFreeze from 'deep-freeze';
+import users from '../../app/assets/javascripts/reducers/users';
+import '../testHelper';
+import {
+  RECEIVE_USERS,
+  ADD_USER,
+  REMOVE_USER,
+  SORT_USERS
+} from '../../app/assets/javascripts/constants';
+
+describe('users reducer', () => {
+  it('should return initial state when no action nor state is provided', () => {
+    const newState = users(undefined, { type: null });
+    expect(newState.users).to.be.an('array');
+    expect(newState.isLoaded).to.eq(false);
+    expect(newState.sort).to.be.an('object');
+  });
+
+  it('returns the previous state and updates users array from action via RECEIVE_USERS, ADD_USER and REMOVE_USER', () => {
+    const initialState = { title: 'title', isLoaded: false };
+    deepFreeze(initialState);
+    const mockedAction = {
+      type: RECEIVE_USERS,
+      data: { course: { users: [1, 2] } }
+    };
+
+    let newState = users(initialState, mockedAction);
+    expect(newState.title).to.eq('title');
+    expect(newState.users).to.deep.eq([1, 2]);
+    expect(newState.isLoaded).to.eq(true);
+
+    mockedAction.type = ADD_USER;
```

Comment:
Reusing the same action and mutating it is bit confusing here, since that's now how actions get used in the app itself. I think making a new, more specifically named action would make this clearer.

Reply:
Should each type be in its own block? It's kinda big chunk for one block on a second look.